### DISCUSSION TITLE
Batch Flop .png in Sub folders under single master directory

### DISCUSSION THREAD

**Discussion by MrAdam1983** [ImageMagick/ImageMagick#5266] [General] [answered]
I have a bunch of .png images that I have organized into various sub-folders (Windows) to help with a later process, but first I need to flop a Duplicate set of the images.

 I know I can Flop images in a single folder but I am not finding what argument I need to make the command look in sub folders. 

This command works in a single folder. 
``magick mogrify -flop *.png"

But I need to add something to the start to get it to look at the subfolder.

**Comment by fmw42** [comment]
Imagemagick will not traverse subdirectories.  You will have to write a script loop over each subdirectory and call magick mogrify for each.

**Comment by MrAdam1983** [comment]
Okay, jives with what else I was seeing. so i came up with this witch executes and shows that it find alls of my images but nothing ever changes. 

`for /R %f in (*) do echo %f magick mogrify -flop *.png`

**Comment by MrAdam1983** [comment, marked as answer]
OMG, I actually figured it out!  At first I didn't think it worked because it threw a bunch of Blob errors about invalid arguments, but then i noticed the images had flipped...

`FORFILES /S /C "magick mogrify -flop *.png"`

**Comment by snibgo** [comment]
You had:
```
echo %f magick mogrify -flop *.png
```
That will echo some text like "filename.ext magick mogrify -flop *.png". It will never execute "magick".

I suppose you want to echo something, and then execute a "magick" command, like this:
```
echo %f magick && mogrify -flop *.png
```
